controllers/certificatesigningrequests_controller.go

```diff
@@ -88,6 +88,15 @@ func (r *certificateSigningRequestsReconciler) Reconcile(ctx context.Context, re
 
 // reconcileCSR handles the CSR validation and approval
 func (r *certificateSigningRequestsReconciler) reconcileCSR(request *certificates.CertificateSigningRequest) error {
+	// If a CSR is approved after being added to the queue, but before we reconcile it,
+	// it may have already been approved. If it has already been approved, trying to
+	// approve it again will result in an error and cause a loop.
+	// Return early if the CSR has been approved externally.
+	if isApproved(request) {
```

Comment:
Does it make sense to take into account if the request has been denied as that would modify the object as well? If yes, can we take advantage of existing method [isPending](https://github.com/openshift/windows-machine-config-operator/blob/6901ac3ed2891b7627a1431ab66703358b902671/controllers/certificatesigningrequests_controller.go#L132)?

Reply:
Yeah that seems reasonable, have updated the PR